### PR TITLE
docs: document logOptions in cilium install

### LINKED ISSUE #5578: Configurable log levels in Cillium are completely undocumented
## Feature Requests

Hi there,

It would be great if it was possible to configure the log level of Cilium.
By default, it looks like Cilium logs at the Info level, with the --debug argument enabling lower levels of logs.

It would be great if this could be more configurable so that, for example, Cilium would only log Warnings and above.

Thanks

---
**Update 2023-11**: We support configurable logging, but it is _entirely undocumented_. That should be fixed.

### PR DESCRIPTION
**Pull request by jjchambl**
Cilium supports overriding default log options, but the default and allowed values for this configuration are undocumented. This PR attempts to provide a structure and specify the defaults for logging options. I have included the json structure only; cilium also supports key/value pairs, but I don't see how it's possible to have both options in the docs effectively -- potentially one reason they were not included to begin with.

Comments/suggestions welcome. I tried to detail the breadth of options as best I could without getting too granular.

Fixes: https://github.com/cilium/cilium/issues/5578

```release-note
Document Cilium's logOptions and overrides
```

### CODE REVIEW

File: install/kubernetes/cilium/values.yaml [hunk 1]
```diff
@@ -1567,9 +1567,20 @@ localRedirectPolicy: false
...
-#   format: json
+# -- (string) Allows you to define logging options.
+# LogOpt sets log driver options for cilium
+# "log-opt", "Log driver options for cilium-health e.g. syslog.level=info,syslog.facility=local5,syslog.tag=cilium-agent"
+# level, network (syslog.LOG_KERN(3)), address, severity(loglevel), facility
+# eg: "level=warn,format=(text|json|json-ts),syslog.facility=local4,syslog.severity=info,syslog.tag=cilium"
+logOptions:
+  level: "info"
+  format: "text"
+  syslog:
```
Comment (gandro) [inline install/kubernetes/cilium/values.yaml]: I don't think we enable the syslog logger by default? Rather we provide an empty driver list, which defaults to stdout: 

https://github.com/cilium/cilium/blob/00ab25294960cf015935303cca68f4720c0b85ed/pkg/logging/logging.go#L165

So I don't think we should provide the syslog options by default either and leave this commented out in the values file.

**Review by gandro — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Please note that our Helm values have a bunch of auto-generated files that also need to be re-genearted as part of this PR.

In particular, please apply your edit to the `values.yaml.tmpl` file and then run the following:

`make -C install/kubernetes`
`make -C Documentation update-helm-values`

And then squash all the changes into a single commit.

The former command will generate the `values.yaml` file and update the README.md documenting the Helm values in the chart, the latter will update the documentation of the Helm values we publish on docs.cilium.io. Thank you.

File: install/kubernetes/cilium/values.yaml (hunk 1 above)
Comment (jjchambl) [reply install/kubernetes/cilium/values.yaml]: > So I don't think we should provide the syslog options by default either and leave this commented out in the values file.

Could we simply point out that syslog is disabled unless configured (I agree, looks like it's disabled by default)? I feel like others could miss that they could configure it at all if it's omitted entirely.

Comment (gandro) [reply install/kubernetes/cilium/values.yaml]: Is there any reason why we can't comment this out, i.e. like this? The syntax is documented with this PR after all?

```yaml
logOptions:
  level: "info"
  format: "text"
  #syslog:
  #  facility: "kernel"
  #  # mirrors "level" setting if none specified
  #  severity: "info"
  #  address: ""
  #  tag: ""
```

File: install/kubernetes/cilium/Chart.yaml-e [hunk 2]
```diff
@@ -0,0 +1,179 @@
+apiVersion: v2
```
Comment (gandro) [inline install/kubernetes/cilium/Chart.yaml-e]: It looks like you committed some unrelated files

**Review by gandro — COMMENTED** [review]
Thanks for fixing up the PR. There seem to be two files that got accidentally committed. Please also make sure to squash your commits into one (e.g using `git rebase`)

**Comment by jjchambl** [comment]
> Thanks for fixing up the PR. There seem to be two files that got accidentally committed. Please also make sure to squash your commits into one (e.g using `git rebase`)

Yeah I messed up the commit history somehow. May need to to a reset and run the make commands again.

**Comment by ti-mo** [comment]
I've converted this to a draft to get this out of triage/review queues. Please mark as ready once you've addressed the outstanding issues.

**Comment by jjchambl** [comment]
Sorry for the delay, my git brain got fried. Will rebase and update the PR shortly.

**Comment by jjchambl** [comment]
I decided to do a hard reset against the main branch since it had been a long time and I messed up the order of the commits earlier.

**Comment by jjchambl** [comment]
Adding signed off message to commit shortly.

**Review by gandro — APPROVED** [review]
Thank you, looks good to me now!

**Comment by gandro** [comment]
The spellchecker is complaining about some of the newly documented words. Please run the following commands to add them to our dictionary:


```
Documentation/update-spelling_wordlist.sh loglevel syslog
```

It is also complaining about the spelling of "eg", perhaps write it as "e.g.". See https://github.com/cilium/cilium/actions/runs/8206745715/job/22503789188?pr=30514

I also think there is a new line missing before the "Signed-off-by" line in your commit message, checkpatch CI is failing https://github.com/cilium/cilium/actions/runs/8206745717/job/22503782540?pr=30514#step:3:43

**Comment by jjchambl** [comment]
> It is also complaining about the spelling of "eg", perhaps write it as "e.g."
I think I originally wrote it as e.g., will just replace with "for example" instead.

Will also update the word list and fix the commit message.

**Comment by gandro** [comment]
Seems the "eg" is still in use, docs workflow is failing: https://github.com/cilium/cilium/actions/runs/8238730356/job/22551996355?pr=30514

File: install/kubernetes/cilium/values.yaml.tmpl [hunk 3]
```diff
@@ -1602,9 +1602,21 @@ localRedirectPolicy: false
 # To include or exclude matched resources from cilium identity evaluation
 # labels: ""
 
-# logOptions allows you to define logging options. eg:
-# logOptions:
-#   format: json
+# -- (string) Allows you to define logging options.
```
Comment (qmonnet) [inline install/kubernetes/cilium/values.yaml.tmpl]: Let's use a consistent style for the option description, “allows you to...”  is not really necessary here.

```suggestion
# -- (string) Define logging options.
```

File: install/kubernetes/cilium/values.yaml.tmpl [hunk 4]
```diff
@@ -1602,9 +1602,21 @@ localRedirectPolicy: false
...
+# LogOpt sets log driver options for cilium
```
Comment (qmonnet) [inline install/kubernetes/cilium/values.yaml.tmpl]: What is `LogOpt`? Don't you mean `logOptions`?

File: install/kubernetes/cilium/values.yaml.tmpl [hunk 5]
```diff
@@ -1602,9 +1602,21 @@ localRedirectPolicy: false
...
+# "log-opt", "Log driver options for cilium-health, for example syslog.level=info,syslog.facility=local5,syslog.tag=cilium-agent"
+# level, network (syslog.LOG_KERN(3)), address, severity(loglevel), facility
+# eg: "level=warn,format=(text|json|json-ts),syslog.facility=local4,syslog.severity=info,syslog.tag=cilium"
```
Comment (qmonnet) [inline install/kubernetes/cilium/values.yaml.tmpl]: Sorry, I don't think this is very clear. Could you please turn this into a proper sentence?

File: install/kubernetes/cilium/values.yaml.tmpl [hunk 6]
```diff
@@ -1602,9 +1602,21 @@ localRedirectPolicy: false
...
+logOptions:
+  level: "info"
+  format: "text"
+  # syslog is disabled by default: specify it to enable
+  # syslog:
+    # facility: "kernel"
+    # mirrors "level" setting if none specified
```
Comment (qmonnet) [inline install/kubernetes/cilium/values.yaml.tmpl]:
```suggestion
    # -- "severity" mirrors logOptions.level by default.
```

File: install/kubernetes/cilium/values.yaml.tmpl [hunk 7]
```diff
@@ -1602,9 +1602,21 @@ localRedirectPolicy: false
...
-#   format: json
+# -- (string) Allows you to define logging options.
+# LogOpt sets log driver options for cilium
+# "log-opt", "Log driver options for cilium-health, for example syslog.level=info,syslog.facility=local5,syslog.tag=cilium-agent"
+# level, network (syslog.LOG_KERN(3)), address, severity(loglevel), facility
+# eg: "level=warn,format=(text|json|json-ts),syslog.facility=local4,syslog.severity=info,syslog.tag=cilium"
+logOptions:
+  level: "info"
+  format: "text"
+  # syslog is disabled by default: specify it to enable
```
Comment (qmonnet) [inline install/kubernetes/cilium/values.yaml.tmpl]: I think the comment should start with a `--` to indicate it refers to the item below, and is not a commented value?

Also, `specify it to enable` is not super clear. Maybe `specify syslog options to enable syslog logging`?

File: install/kubernetes/cilium/values.yaml.tmpl [hunk 8]
```diff
@@ -1602,9 +1602,21 @@ localRedirectPolicy: false
 # To include or exclude matched resources from cilium identity evaluation
 # labels: ""
 
-# logOptions allows you to define logging options. eg:
-# logOptions:
-#   format: json
+# -- (string) Allows you to define logging options.
+# LogOpt sets log driver options for cilium
+# "log-opt", "Log driver options for cilium-health, for example syslog.level=info,syslog.facility=local5,syslog.tag=cilium-agent"
```
Comment (qmonnet) [inline install/kubernetes/cilium/values.yaml.tmpl]: Is it “for cilium” or “for cilium-health”?

**Review by qmonnet — CHANGES_REQUESTED** [review]
Thanks for this! I see several things to improve, please see below.

I would also recommend to state the motivation and context in your commit description: what you currently have in the PR description could also be reported in the commit description, where it's easier to find when people look at the Git history.

File: install/kubernetes/cilium/values.yaml.tmpl (hunk 4 above)
Comment (jjchambl) [reply install/kubernetes/cilium/values.yaml.tmpl]: Yes, logOptions. It's named/called differently in other parts of the code, another copy/pasta issue. I'll correct this.

File: install/kubernetes/cilium/values.yaml.tmpl (hunk 8 above)
Comment (jjchambl) [reply install/kubernetes/cilium/values.yaml.tmpl]: it's for cilium, though cilium-health inherits it. a few of these problems are from me copy-pastaing and not cleaning it up. will try to get to it today.

File: install/kubernetes/cilium/values.yaml.tmpl [hunk 9]
```diff
@@ -1602,9 +1602,18 @@ localRedirectPolicy: false
 # To include or exclude matched resources from cilium identity evaluation
 # labels: ""
 
-# logOptions allows you to define logging options. eg:
-# logOptions:
-#   format: json
+# -- (string) Define logging options.
+# These options can also be set as json key value pairs, for example "level=warn,format=(text|json|json-ts),syslog.facility=local4,syslog.severity=info,syslog.tag=cilium"
```
Comment (qmonnet) [inline install/kubernetes/cilium/values.yaml.tmpl:1606]: Can you please keep a single example for the `format` and remove the pipes (`|`)? It's not your fault, but it turns out they're breaking the rendering and I see no trivial way to escape them. See how the end of the sentence is missing in the generated helm-values.rst file. Let's add a comment above the `format:` option below to document the existing values instead?

**Review by qmonnet — CHANGES_REQUESTED** [review]
Looks nearly all good, thanks for the changes!

It turns out there's an issue with the pipe characters in the description though, please see below.

**Review by joamaki — APPROVED** [review]
